ds/ds.go

```diff
@@ -357,6 +358,67 @@ func Lookup(query dsQuery) (string, string, error) {
 	return e.IPs[0].String(), strconv.Itoa(e.Port), nil
 }
 
+// LookupResult is returned over a channel, and has either a result or an error.
+type LookupResult struct {
+	Entry *dnssd.BrowseEntry
+	Error error
+}
+
+// LookupN finds a number of hosts and returns a slice of *LookupResult and an error, if any.
+func LookupN(query dsQuery, n int) ([]*LookupResult, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), dsTimeout)
+	context.Canceled = fmt.Errorf("LookupChan:%w", syscall.ECANCELED)
+	context.DeadlineExceeded = fmt.Errorf("LookupChan:%w", syscall.ETIMEDOUT)
+	service := fmt.Sprintf("%s.%s.", query.Type, query.Domain)
+	defer cancel()
+
+	v("Browsing for %s\n", service)
+
+	respCh := make(chan *LookupResult, n)
+
+	addFn := func(e dnssd.BrowseEntry) {
+		v("%s	Add	%s	%s	%s	%s (%s)\n", time.Now().Format(timeFormat), e.IfaceName, e.Domain, e.Type, e.Name, e.IPs)
+		// check requirement
+		v("Checking %q, %q", e.Text, query.Text)
+		if required(e.Text, query.Text) {
+			if (query.Instance != "") && (e.ServiceInstanceName() != query.Instance+"."+service) {
+				v("Instance %s didn't match %s", e.ServiceInstanceName(), query.Instance+"."+service)
+			} else {
+				respCh <- &LookupResult{Entry: &e}
+			}
+		}
+	}
+
+	rmvFn := func(e dnssd.BrowseEntry) {
+		v("%s	Rmv	%s	%s	%s	%s\n", time.Now().Format(timeFormat), e.IfaceName, e.Domain, e.Type, e.Name)
+		// we aren't maintaining cache so don't care?
+	}
+
+	go func() {
+		// A timeout will occur at some point, whether any results have been delivered or not.
+		// Best to just close the channel with whatever was provided.
+		if err := dnssd.LookupType(ctx, service, addFn, rmvFn); !errors.Is(err, syscall.ETIMEDOUT) {
+			v("Adding lookupresult with err %v %T", err, err)
+			respCh <- &LookupResult{Error: err}
+		}
+		close(respCh)
+	}()
+
+	var ret []*LookupResult
+	var err error
+	for l := range respCh {
+		if l.Error != nil {
+			err = l.Error
+			break
+		}
+		ret = append(ret, l)
+		if len(ret) >= n {
+			break
+		}
+	}
+	return ret, err
```

Comment:
Wouldn't we just need the dsSort before return like in standard Lookup() to get the sorted responses?

Reply:
The problem I have with that is -- sorted by what key? I'm a bit leery of embedding a sort in here without knowing what the user wants to sort on?

Reply:
anyway ... I view the sort as another PR if you don't mind, makes bisect better

Reply:
btw the idiomatic way to do this ... you'd change it as follows:
type Opt func([]*LookupResult) ([]*LookupResult, error)

func LookupN(query dsQuery, n int, opts ...Opt) ([]*LookupResult, error) {

right before the return:
for _, o := range opts {
   ret, err = o(ret)
   if err != nil {
    return ret, err
}

}

That lets you refine in any way you choose.
I'd rather do that than wire in a sort anyway.

Reply:
So ... here's the thing.  dsSort was already in the code as part of Lookup, you are essentially removing the sort functionality by switching to LookupN as the default when switching to LookupN, but you don't even change the README to remove the sort options nor do you remove the sort tests in TESTDOCKERCPU -- so its actually you that is breaking the bisect.

Reply:
LookupN was not in the package before, and the sorts were deadcode.

I'll close this PR, go ahead and do what you think best, but as it is it's hard to get "more than 1" from a lookup.